### ISSUE TITLE
Refactor the annotation used by the DSL

### ISSUE THREAD

**Issue by pith** [seedstack/business#24]
Change this annotation:

``` java
public @interface MatchingFactoryParameter {

    int index() default -1;

    int typeIndex() default -1;

}
```

to

``` java
public @interface MatchFactoryParam {

    int index() default -1;

    int aggregateIndex() default -1;

}
```

The annotation resolver should be changed according to this modification. The same modification should be applied to the `@MatchingEntityId` -> `@MatchId`. Also think to the possibility to change index into value...

The previous annotations and resolver should be kept for backward compatibility.

**Comment by adrienlauer** [comment]
Related to #23

**Comment by pith** [comment]
I closed the #23 and report it here:
- Think to add the possibility to remove the index when the parameter types are not ambiguous.
- Rename Annotation with shorter names.

**Comment by adrienlauer** [comment]
Done, without adding the possibility to remove the index (would be too unreliable).
